File: Programmers/LEVEL_2/c++/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> temp;

    int max = 0;
    for (int i = 0; i < progresses.size(); i++)
    {
        int quo = (100 - progresses[i]) / speeds[i];
        int remain = (100 - progresses[i]) % speeds[i];

        if (remain != 0) quo++;

        temp.push_back(quo);

        if (max < quo)
        {
            answer.push_back(1);
            max = quo;
        }
        else
        {
            int temp = answer.back();
            answer.pop_back();
            answer.push_back(temp + 1);
        }
    }

    return answer;
}

File: Programmers/LEVEL_2/기능개발.js
function solution(progresses, speeds) {
  var answer = [];

  let max = 0;
  let temp = [];
  for (let i = 0; i < progresses.length; i++) {
    let quo = Math.ceil((100 - progresses[i]) / speeds[i]);

    if (max < quo) {
      answer.push(1);
      max = quo;
    } else {
      let temp = answer.pop();
      answer.push(temp + 1);
    }
  }

  return answer;
}
